src/translate/index.ts

```diff
@@ -148,6 +151,16 @@ export default async function translate(this: Command) {
 
       meta[metaKey] = meta[metaKey] ?? {};
 
+      const contextToAdd = typeof sourceText === "string" ? sourceText : Object.values(sourceText).join('\n');
+
+      if(contextBuffer.length > maxContextLength) {
+        let newBuffer = contextBuffer.substring(contextToAdd.length);
+        newBuffer += contextToAdd;
+        contextBuffer = newBuffer;
+      } else {
+        contextBuffer += contextToAdd;
+      }
+
```

Comment:
@genox do I understand it correct, that the context will just receive the source value and that's enough?

I actually thought, that the context is more like a standard prompt to tell deepl about the translation.
Do you have a link to the documentation I looked it up, but wasn't able to find it.

Reply:
https://www.deepl.com/de/docs-api/translate-text/translate-text

Look for the context request parameter.

It's not a prompt. What you do when you provide context is that the requested string is translated in a context - like each new sentence would be in a larger document. Deepl splits longer texts into sentences and translates them one by one. The entirety of the text is each operations context.

They do write it isn't production ready, but well.. what to do - every other solution is much more complicated.

Reply:
Ah I see. This looks good I will test it on the weekend and merge/deploy it.

I will play around with it. We have already 500+ translations in our qwik app.